admiral/pkg/clusters/handler.go

```diff
@@ -483,15 +484,16 @@ func handleVirtualServiceEvent(
 					log.Infof(LogFormat, "Delete", "VirtualService", obj.Name, clusterId, "Success")
 				}
 			} else {
-				exist, _ := rc.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(syncNamespace).Get(ctx, obj.Name, v12.GetOptions{})
+				exist, _ := rc.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(syncNamespace).Get(ctx, obj.Name, metav1.GetOptions{})
+				// nolint
 				addUpdateVirtualService(ctx, obj, exist, syncNamespace, rc)
 			}
 		}
 	}
 	return nil
 }
 
-func addUpdateVirtualService(ctx context.Context, obj *v1alpha3.VirtualService, exist *v1alpha3.VirtualService, namespace string, rc *RemoteController) {
+func addUpdateVirtualService(ctx context.Context, obj *v1alpha3.VirtualService, exist *v1alpha3.VirtualService, namespace string, rc *RemoteController) error {
```

Comment:
Can we add a function to evaluate if the update is destructive or not? 

And do we need knowledge from multiple clusters to build this virtual service? Just like admiral requires knowledge from multiple clusters to build service entries, hence we skip destructive updates during start up time.

Reply:
Add some logging on diff